### DISCUSSION TITLE
Why is there only 4 threads running when there are 1268 threads?

### DISCUSSION THREAD

**Discussion by bukunmilab** [postalserver/postal#3166] [Help with using Postal]
I am trying to maximize the my sending speed on a 12 GB RAM server. So I ran the `htop` command in SSH. I have only Postal and Redis running on the server and here is what the server looks like:

<img width="752" alt="Screenshot_1" src="https://github.com/user-attachments/assets/239a5784-bbce-45cd-96ba-9b36273c83a8">

I use the SSH command: `postal start --scale worker=20`
But my sending speed is the same even with 20 or 30 or 40 workers and the value of "thr running" is never more than 4. What is the reason for this?

From a 2GB CPU core , 6 GB RAM, 30GB SSD, I upgraded to this:
**VPS specs:**
RAM: 12 GB
CPU Cores: 4 GB
NVme SSD: 50 GB

Yet my sending speed is still only 130- 200 per minute now. 
I have already played around with the my.cnf file and postal.yml.
What can I tune or tweak to bring it to at least 350-500 emails per minute?

**Comment by willpower232** [comment]
> the value of "thr running" is never more than 4

I'm not sure what you mean by this

It looks like you have made all four cores super busy with all the messages you're trying to send so not sure what else you can do. Where are you running your database? Perhaps you could set up a local connection between the two servers to speed up the network connections a litte.

**Reply by bukunmilab** [reply]
Now I understand. I also think connection speed plays a key role in this. 
Can you please share a reference on how to setup a local connection between two server?

**Reply by willpower232** [reply]
If both servers are with the same provider and set up next to each other in a VPC or similar, then you could probably use a local IP address instead of the public internet IP address.
